util/testutil/dockerd/client/options.go

```diff
@@ -0,0 +1,35 @@
+package client
+
+import (
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Opt is a configuration option to initialize a [Client].
+type Opt func(*Client) error
+
+// WithHost overrides the client host with the specified one.
+func WithHost(host string) Opt {
+	return func(c *Client) error {
+		hostURL, err := ParseHostURL(host)
+		if err != nil {
+			return err
+		}
+		c.host = host
+		c.proto = hostURL.Scheme
+		c.addr = hostURL.Host
+		c.basePath = hostURL.Path
+		if tr, ok := c.client.Transport.(*http.Transport); ok {
```

Comment:
This doesn't seem to do anything based on host, so can be in the default constructor.

Reply:
I tried changing this but it does seem to matter. It does things based on the proto and host and has been the cause of the new failures so I'm going to revert this back to what's in the original client.